### ISSUE TITLE
repo: filters are difficult to use

### ISSUE THREAD

**Issue by sqs** [sourcegraph/sourcegraph#130]
A user at GopherCon (will mention pending permission) mentioned that repo: filters are difficult to use because they use regexps. *This issue is a placeholder for me to gather more info on this feedback.*

Possible solutions to investigate/suggest:

### Dynamic search filters

Sourcegraph suggests search filters based on the results. See the `Repositories:` line underneath the search box. Clicking on one of those repositories filters the search results to only those from that repository (by adding a `repo:` filter to the query).

![screenshot from 2018-08-28 00-29-50](https://user-images.githubusercontent.com/1976/44704675-7cf45a00-aa59-11e8-903a-6e69875bd75e.png)
https://sourcegraph.com/search?q=repo:dgryski/+open

### More lenient regexp parsing

Some users have requested that Sourcegraph be lenient/smart when parsing regexps, so that (for example)

- `myfunc(` would be interpreted as `myfunc\(`
- `file:.go` would be interpreted as `file:\.go$` (ie, filenames ending in `.go`)

We could extend this to repositories, so that `repo:foo` would be interpreted smartly as `repo:/foo$` or perhaps incorporating some default org name (or ranking repos from certain orgs above others).

**Comment by felixfbecker** [comment]
One thing we could do is have search suggestions return an error when the query is invalid and display that error in the suggestions box (i.e. "Invalid regular expression for `repo:` filter, unexpected (")

Another thing would be syntax highlighting.

**Comment by keegancsmith** [comment]
I've seen repo: filters which are just `strings.Contains`. However, I've found it annoying when wanting to limit to a repo who's name is a prefix of others. Another way which would likely click with developers is to use globs instead (similarly for writing the file: queries)

If we got better at ranking repos, would that solve this particular frustration (possibly mixed with dynamic query filters)?
